tests/backend_/test_base.py

```diff
@@ -21,3 +22,23 @@ def test_set_autocommit(self):
         self.assertIs(connection.get_autocommit(), False)
         connection.set_autocommit(True)
         self.assertIs(connection.get_autocommit(), True)
+
+    def test_connection_created_database_attr(self):
+        """
+        connection.database is available in the connection_created signal.
+        """
+        data = {}
+
+        def receiver(sender, connection, **kwargs):  # noqa: ARG001
+            data["database"] = connection.database
+
+        connection_created.connect(receiver)
+        connection.close()
```

Comment:
Can we also assert that data is empty after calling `close()` or is that not necessary?

Reply:
Certainly could, but if `close()` actually created a connection (thus firing the `connection_created` signal) instead of closing, I think there would be other rather obvious problems.